### DISCUSSION TITLE
OTLP | Absl, GSL ,STL settings

### DISCUSSION THREAD

**Discussion by yonatan555** [open-telemetry/opentelemetry-cpp#1773] [Q&A]
Hi , I tried to understand what exactly are stl, abseil, gsl , macros? 
Can you please elaborate about this, I did some research on it ib Google, didn't get understand thr full idea

If i want to build an open telemetry projevt in centos 7 OS , C++14, 9.3.1 compiler, which of these variables needs to be set?

( I want to support otlp_grpc gRPC , Prometheus , 
 metrics too.) What cmake command should be,(i want to build grpc in separate too and corporate this in otlp project)

**Comment by esigo** [comment]
Hi @yonatan555, thanks for reaching out to us.
Please see an example for centos [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docker/Dockerfile.centos)

**Reply by yonatan555** [reply]
Ok, thanks for your help, do you think is it enough to srtart with grpc,otlp, and Prometheus

**Reply by esigo** [reply]
only `-DWITH_PROMETHEUS=ON` is missing. You also need to install Prometheus as is done https://github.com/open-telemetry/opentelemetry-cpp/blob/a3e3a90ec6981d85375addb99ffac898341f144f/ci/do_ci.sh#L8-L18

**Reply by yonatan555** [reply]
Ok thank you very much, I will update ASAP

**Comment by yonatan555** [comment]
Hi ,the Grpc is failed in this command:
cmake3 \
        -DCMAKE_CXX_STANDARD=17 \
        -DCMAKE_BUILD_TYPE=Release \
        -DgRPC_INSTALL=ON \
        -DgRPC_BUILD_TESTS=OFF \
        -DgRPC_ABSL_PROVIDER=package \
        -DgRPC_SSL_PROVIDER=package \
        .. && \

I dont know why
